ansible/roles/sonicv2/templates/etc/systemd/system/swss.j2

```diff
@@ -5,6 +5,8 @@ After=database.service
 
 [Service]
 User={{ sonicadmin_user }}
+# Wait for redis server start before database clean by checking the server listening port 6379
+ExecStartPre=/bin/bash -c "while true; do if [ -n \"$(netstat -l | grep 6379)\" ]; then break; fi; sleep 1; done"
```

Comment:
@lguohan netsat output looks as follows on the box:

admin@switch2:~$ netstat -l
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State      
tcp        0      0 localhost:6379          *:*                     LISTEN 

So I guess it's better to check for ":6379". The it will cover all the cases.